### DISCUSSION TITLE
Rich Text Format (RTF) in Quest QuestPDF

### DISCUSSION THREAD

**Discussion by appwarehouse** [QuestPDF/QuestPDF#276] [General]
Hello all, I'm hoping you can help with a little problem I have.

I've got a situation where I'm attempting to generate PDF Documents using data stored in a database. The format of the content is Rich Text Format (RTF) this has been done to preserve the formatting of the data each time it is saved.

Has anyone used rtf to populate their documents or can anyone offer ideas of how to successfully do this while preserving the formatting of the text.

**Comment by Serverraum** [comment]
Hm, as I found this question whilst also searching for possible solutions, I would try the following:
Parse  RTF to HTML and then use the HTMLtoQPDF AddIn for QuestPDF.

https://github.com/Relorer/HTMLToQPDF

That should be the easiest solution.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response.

QuestPDF does not natively support RTF parsing, as it is a PDF layout engine rather than a document format converter. However, as @Serverraum suggested, a practical approach would be to first convert your RTF content to HTML using a parser of your choice, and then use the community-maintained [QuestPDF.HTML](https://www.nuget.org/packages/QuestPDF.HTML) NuGet package ([HTMLToQPDF](https://github.com/Relorer/HTMLToQPDF)) to render it within your QuestPDF documents.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
